tcms/rpc/api/testcase.py

```diff
@@ -279,8 +281,20 @@ def filter(query=None):  # pylint: disable=redefined-builtin
     if query is None:
         query = {}
 
-    return list(
-        TestCase.objects.filter(**query)
+    # Cannot use `setup_duration` and `testing_duration` as aliases because Django
+    # doesn't allow annotating querysets with aliases which match existing fieldnames.
+    # https://code.djangoproject.com/ticket/11256
+    # https://github.com/django/django/blob/main/django/db/models/query.py#L1154-L1157
+
+    # we're adding the suffix `_ms` to the aliases because they are
+    # `Cast`ed to their millisecond values
+    qs = (
+        TestCase.objects.annotate(
+            setup_duration_sec=Cast("setup_duration", IntegerField()) / 10 ** 6,
+            testing_duration_sec=Cast("testing_duration", IntegerField()) / 10 ** 6,
+            expected_duration_sec=Coalesce("setup_duration_sec", 0) + Coalesce("testing_duration_sec", 0),
+        )
```

Comment:
This isn't going to work because apparently Postgres can't cast intervals to double (float) and working on Postgres is pretty important for us. 

I've added a patch to the XML-RPC and JSON-RPC serializers which converts all timedelta values to seconds (float) without changing the field names, see https://github.com/kiwitcms/Kiwi/pull/2542.

Adjust the query and leave the tests. 

Note: computing expected_duration via the SQL query will probably still fail for Postgres. If that's the case leave this field out of the returned values. It can be easily calculated on the client side.